### PR TITLE
(Core/Quest) Refresh SMSG_QUESTGIVER_STATUS_MULTIPLE after a quest change

### PR DESCRIPTION
**Pull request by acidmanifesto**
This updates the status of nearby quest givers and prevents them from being stuck in a state that's no longer valid (for example, stuck with yellow question mark until area update).
This issue was discovered and was made in support of Malcrom's Quest Scripting https://github.com/azerothcore/azerothcore-wotlk/pull/8895 where it became even more fraustrating.

This will force the quests status to update while being in the same area without the need to relog or re-enter the area id.

## Changes Proposed:
-  Move SendQuestGiverStatusMultiple
-  Create SendQuestGiverStatusMultiple

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8800
- Supports PR https://github.com/azerothcore/azerothcore-wotlk/pull/8895 and various other quests.

## Tests Performed:
Builds and compiles with no issues. Tested and seems to work for me now.

## How to Test the Changes:

1. Best and quickest way to perform this is to find a quest giver that is both a start and turn in of a quest.
2. Take the quest from the quest giver natuarlly
3. Do not leave 
4. Do a .quest complete of the said quest, notice
5. You will notice the ! will go from yellow to grey ? on acceptance, and will then turn yellow when quest complete is done.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
@Nyeriah Review me please.

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -7438,7 +7440,51 @@ bool Player::CheckAmmoCompatibility(const ItemTemplate* ammo_proto) const
...
+            data << uint8(questStatus);
+            ++count;
+        }
+    }
+
+    data.put<uint32>(0, count); // write real count
+    GetSession()->SendPacket(&data);
+}
+
+    /*  If in a battleground a player dies, and an enemy removes the insignia, the player's bones is lootable
```
Comment (Nyeriah) [inline src/server/game/Entities/Player/Player.cpp:7487]:
```suggestion
/*  If in a battleground a player dies, and an enemy removes the insignia, the player's bones is lootable
```

**Comment by Nyeriah** [comment]
> @Nyeriah Review me please.

Just the ident here: https://github.com/azerothcore/azerothcore-wotlk/pull/8902/files#r741294069 seems to be off on my display, check if looks okay on yours

Rest looks fine for me, just needs to be tested in-game

**Review by Nyeriah — APPROVED** [review]

File: src/server/game/Entities/Player/Player.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/game/Entities/Player/Player.cpp]: You are right. There is a unneeded indent there

**Comment by acidmanifesto** [comment]
@azerothcore/testers

**Comment by malcrom** [comment]
Fixes issue #8800

**Comment by tkn963** [comment]
I can't test this because I can't seem to trigger the issue at hand.

Without using a PR, I set my level to 15, went to The Crossroads and accepted quests from NPCs as you normally would, ran a .quest complete, and always looked at the icon above their heads. I also tried this on some quests in other areas. They always updated as they should for me.

I would appreciate some info on a situation where the issue often occurs so I can test the PR.

**Comment by acidmanifesto** [comment]
> I can't test this because I can't seem to trigger the issue at hand.
> 
> Without using a PR, I set my level to 15, went to The Crossroads and accepted quests from NPCs as you normally would, ran a .quest complete, and always looked at the icon above their heads. I also tried this on some quests in other areas. They always updated as they should for me.
> 
> I would appreciate some info on a situation where the issue often occurs so I can test the PR.

This PR quest that malcrom fixed or start halloween event and do the jack o lanterns as malcrom stated in https://github.com/azerothcore/azerothcore-wotlk/issues/8800

**Comment by acidmanifesto** [comment]
.start event 12 and then click on the jackolatern and hit complete. the ! will go away. **Before the PR** it will not go away unless you re enter the area id or relog.
https://youtu.be/EcNSIcRHuMA

**Comment by tkn963** [comment]
> 
> 
> .start event 12 and then click on the jackolatern and hit complete. the ! will go away. **Before the PR** it will not go away unless you re enter the area id or relog. Before the PR: https://youtu.be/LoLCV1x7vwM After the PR: https://youtu.be/EcNSIcRHuMA

That I knew of, noticed a lot the lanterns having ! no matter what I did. Some had ? and some had !

At least I know how to test it now.

I was confused because the PR steps start with "Best and quickest way to perform this is to find a quest giver that is both a start and turn in of a quest."

**Comment by acidmanifesto** [comment]
> > .start event 12 and then click on the jackolatern and hit complete. the ! will go away. **Before the PR** it will not go away unless you re enter the area id or relog. Before the PR: https://youtu.be/LoLCV1x7vwM After the PR: https://youtu.be/EcNSIcRHuMA
> 
> That I knew of, noticed a lot the lanterns having ! no matter what I did. Some had ? and some had !
> 
> At least I know how to test it now.
> 
> I was confused because the PR steps start with "Best and quickest way to perform this is to find a quest giver that is both a start and turn in of a quest."

I apologize sincerely for the confusion. I been pulling some late nighters so my congnitivity both shows in my messages and prs. haha. Thank you for your patience.

**Comment by tkn963** [comment]
@acidmanifesto That's perfectly okay, it got cleared up in the end.

It works as it should, tested with a lot of pumpkins and there were no issues. 👍

**Pull request merged**